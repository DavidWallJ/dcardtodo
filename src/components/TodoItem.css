.todoItem {
	text-align: left;
}

.todoTextSpan {
	font-size: 17px;
}

.todoItem {
	display: flex;
	justify-content: space-between;
	align-items: center;
}

.todoItemRight {
	justify-content: space-between;
}

.todoItem input[type='checkbox'] {
	margin-top: 6px;
}

.editTodoModal {
	margin-top: 13%;
}

.modalContainer {
	display: flex;
	justify-content: space-between;
	align-items: center;
}

.modalContainer h4 {
	margin: 10px;
}

.modalCheckButton {
	margin: 10px;
}
