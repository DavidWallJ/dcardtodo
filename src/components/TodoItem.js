import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { connect } from 'react-redux';
import './TodoItem.css';
import { Checkbox, Button, Modal, FormControl } from 'react-bootstrap';
import moment from 'moment';
import * as actions from '../actions/actions';
import FaPencil from 'react-icons/lib/fa/pencil';

class TodoItem extends Component {
	constructor() {
		super();

		this.state = {
			showModal: false,
			editTodoInput: 'buy some beer'
		};
	}

	close() {
		this.setState({ showModal: false });
	}

	open() {
		this.setState({ showModal: true });
	}

	handleOnChange = () => {
		const { id } = this.props;
		this.props.toggleTodo(id);
	};

	handleOnClick = () => {
		this.open();
		console.log(this.props.text);
	};
	renderDate = (createdAt, completedAt, completed) => {
		let message = 'Created: ';
		let timestamp = createdAt;

		if (completed) {
			message = 'Completed: ';
			timestamp = completedAt;
		}

		return message + moment.unix(timestamp).format('MMM Do YYYY @ h:mm a');
	};
	render() {
		const { text, completed, createdAt, completedAt } = this.props;
		return (
			<div className="todoItem">
				<div className="todoItemLeft">
					<Checkbox onChange={this.handleOnChange} checked={completed}>
						<span className="todoTextSpan">{text}</span>
						<br /> {this.renderDate(createdAt, completedAt, completed)}
					</Checkbox>
				</div>
				<div className="todoItemRight">
					<Button onClick={this.handleOnClick.bind(this)}>
						<FaPencil />
					</Button>
				</div>
				<Modal show={this.state.showModal} onHide={this.close}>
					<Modal.Header>
						<Modal.Title>Edit Todo</Modal.Title>
					</Modal.Header>
					<Modal.Body>
						<FormControl
							className="editTodoInput"
							type="text"
							value={this.state.editTodoInput}
							onChange={event =>
								this.setState({ editTodoInput: event.target.value })}
						/>
					</Modal.Body>
					<Modal.Footer>
						<Button onClick={this.close.bind(this)}>Close</Button>
					</Modal.Footer>
				</Modal>
			</div>
		);
	}
}

export default connect(null, actions)(TodoItem);
