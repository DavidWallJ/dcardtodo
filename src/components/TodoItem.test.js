import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoItem from './TodoItem';

describe(TodoItem, () => {
	const component = shallow(<TodoItem />);

	it('should exist', () => {
		const component = renderer.create(<TodoItem />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
