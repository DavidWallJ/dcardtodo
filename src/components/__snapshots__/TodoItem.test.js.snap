// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class TodoItem extends _react.Component {
	render() {
		const { text, id } = this.props;
		return (
			_react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 7 }, __self: this },
				_react2.default.createElement('p', { __source: { fileName: _jsxFileName, lineNumber: 8 }, __self: this },
					id, '. ', text)));



	}} should exist 1`] = `
<div>
  <p>
    . 
  </p>
</div>
`;
